src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/PartyAddons.kt

```diff
@@ -0,0 +1,228 @@
+        } else if (message == "§9§m-----------------------------------------------------§r" && awaitingMessage != 0) {
+            awaitingMessage--
+            if (awaitingMessage == 1 || party.isEmpty()) return
+
+            val component = UMessage("§aParty members (${party.size})\n")
+
+            val player = party.first { it.name == Skytils.mc.thePlayer.name }
+
+            if (player.type == PartyMemberType.LEADER) {
+                component.append(
+                    createButton(
+                        "§9[Party Warp] ",
+                        "/p warp",
+                        "§9Click to warp the party."
+                    )
+                ).append(
+                    createButton(
+                        "§e[Toggle All Invite]\n",
+                        "/p settings allinvite",
+                        "§eClick to toggle all invite."
+                    )
+                ).append(
+                    createButton(
+                        "§6[Toggle Mute] ",
+                        "/p mute",
+                        "§6Click to toggle mute."
+                    )
+                ).append(
+                    createButton(
+                        "§c[Kick Offline] ",
+                        "/p kickoffline",
+                        "§cClick to kick offline members."
+                    )
+                ).append(
+                    createButton(
+                        "§4[Disband]\n",
+                        "/p disband",
+                        "§4Click to disband the party."
+                    )
+                )
+            }
+
+            val partyLeader = party.first { it.type == PartyMemberType.LEADER }
+            component.append(
+                "\n${partyLeader.status}➡§r ${partyLeader.rank}${partyLeader.name}"
+            )
+
+            if (party.any { it.type == PartyMemberType.MODERATOR }) component.append("\n§eMods")
+            party.filter { it.type == PartyMemberType.MODERATOR }.forEach {
+                component.append(
+                    "\n${it.status}➡§r ${it.rank}${it.name} "
+                )
+                if (player.type != PartyMemberType.LEADER) return@forEach
+                component.append(
+                    createButton(
+                        "§a[⋀] ",
+                        "/p promote ${it.name}",
+                        "§aPromote ${it.name}"
+                    )
+                ).append(
+                    createButton(
+                        "§c[⋁] ",
+                        "/p demote ${it.name}",
+                        "§cDemote ${it.name}"
+                    )
+                ).append(
+                    createButton(
+                        "§4[☠]",
+                        "/p kick ${it.name}",
+                        "§4Kick ${it.name}"
+                    )
+                )
+            }
+
+            if (party.any { it.type == PartyMemberType.MEMBER }) component.append("\n§eMembers")
```

Comment:
iterates through the list twice for no reason, maybe save the result of the filter

Reply:
val mods = party.filter(...)
if (mods.isNotEmpty()) "Mods"
assuming something like this right?